src/NHibernate/ISynchronizableQuery.cs

```diff
@@ -0,0 +1,34 @@
+using System.Collections.Generic;
+
+namespace NHibernate
+{
+	public interface ISynchronizableQuery<out T> where T : ISynchronizableQuery<T>
```

Comment:
Do we plant to have other types of `ISynchronizableQuery<>` other than `ISQLQuery`?

Reply:
Looking at Hibernate code, it seems not. They use it for `SQLQuery` (obsoleted on their side) and `NativeQuery` (replacement for `SQLQuery`). So why not merging that in `ISynchronizableSQLQuery` without the generic param trick.